main/src/ca/uwaterloo/flix/language/phase/Kinder.scala

```diff
@@ -167,11 +167,12 @@ object Kinder {
     * Performs kinding on the given enum case under the given kind environment.
     */
   private def visitCase(caze0: ResolvedAst.Declaration.Case, tparams: List[KindedAst.TypeParam], resTpe: Type, kenv: KindEnv, taenv: Map[Symbol.TypeAliasSym, KindedAst.TypeAlias], root: ResolvedAst.Root)(implicit sctx: SharedContext, flix: Flix): KindedAst.Case = caze0 match {
-    case ResolvedAst.Declaration.Case(sym, tpe0, loc) =>
-      val t = visitType(tpe0, Kind.Star, kenv, taenv, root)
+    case ResolvedAst.Declaration.Case(sym, tpes0, loc) =>
+      val ts = tpes0.map(visitType(_, Kind.Star, kenv, taenv, root))
       val quants = tparams.map(_.sym)
-      val sc = Scheme(quants, Nil, Nil, Type.mkPureArrow(t, resTpe, sym.loc.asSynthetic))
-      KindedAst.Case(sym, t, sc, loc)
+      val schemeBase = if (ts.nonEmpty) Type.mkPureUncurriedArrow(ts, resTpe, sym.loc.asSynthetic) else resTpe
```

Comment:
Type.mkPureUncurriedArrow makes a nonsense type for empty arguments. I could also make it return a non-arrow for empty arguments so I don't have to check multiple places?